OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/cg/AndroidICCAnalysis.scala

```diff
@@ -0,0 +1,1413 @@
+    var dataPaths: ListBuffer[String] = ListBuffer.empty[String]
+    var dataSchemes: ListBuffer[String] = ListBuffer.empty[String]
+    var dataSSPs: ListBuffer[String] = ListBuffer.empty[String]
+
+    def addAction(action: String): Unit = {
+        actions += action
+
+    }
+    def addCategory(category: String): Unit = {
+        categories += category
+    }
+    def addDataType(dataType: String): Unit = {
+        dataTypes += dataType
+    }
+
+    def setReceiver(rec: ClassFile): Unit = {
+        receiver = rec
+    }
+
+    def cloneFilter(): intentFilter = {
+        val clone = new intentFilter()
+        clone.actions = actions
+        clone.categories = categories
+        clone.dataAuthorities = dataAuthorities
+        clone.dataTypes = dataTypes
+        clone.dataPaths = dataPaths
+        clone.dataSchemes = dataSchemes
+        clone.componentType = componentType
+        clone.receiver = receiver
+        clone
+    }
+
+    def evaluateUseSites(
+                          tacCode: AITACode[TACMethodParameter, ValueInformation],
+                          useSites: List[Int]
+    ): Unit = {
+        val statements = tacCode.stmts
+        useSites.foreach { ln ⇒
+            val stm = statements(ln)
+            if (stm.isVirtualMethodCall) {
+                val name = stm.asVirtualMethodCall.name
+
+                val parameter = stm.asVirtualMethodCall.params
+                if (name == addAction) {
+                    actions += parameter.head.asVar.value.toString.replaceAll(paramRegex, "")
+                }
+                if (name == addCategory) {
+                    categories += parameter.head.asVar.value.toString.replaceAll(paramRegex, "")
+
+                }
+                if (filterDataMethods.contains(name)) {
+                    name match {
+                        case `addDataAuthority` ⇒ dataAuthorities += (parameter.head.asVar.value.toString.
+                            replaceAll(paramRegex, "") + parameter.last.asVar.value.toString.
+                            replaceAll(paramRegex, ""))
+                        case `addDataPath` ⇒ dataPaths += (parameter.head.asVar.value.toString.
+                            replaceAll(paramRegex, "") + parameter.last.asVar.value.toString.
+                            replaceAll(paramRegex, ""))
+                        case `addDataScheme` ⇒ dataSchemes += parameter.head.asVar.value.toString.
+                            replaceAll(paramRegex, "")
+                        case `addDataSSP` ⇒ dataSSPs += parameter.head.asVar.value.toString.
+                            replaceAll(paramRegex, "")
+                        case `addDataType` ⇒ dataTypes += parameter.head.asVar.value.toString.
+                            replaceAll(paramRegex, "")
+                    }
+                }
+            }
+            if (stm.isAssignment) {
+                if (stm.asAssignment.expr.asVirtualFunctionCall.name == registerReceiver) {
```

Comment:
Looks like an exception waiting to happen. What if the expr isn't a VirtualFunctionCall?